src/lib/PublicRequestTypes.ts

```diff
@@ -139,20 +130,58 @@ export interface RenameRequest extends SimpleRequest {
     address?: string; // Userfriendly address
 }
 
+export enum CashlinkState {
+    UNKNOWN = -1,
+    UNCHARGED = 0,
+    CHARGING = 1,
+    UNCLAIMED = 2,
+    CLAIMING = 3,
+    CLAIMED = 4,
+}
+
+export enum CashlinkTheme {
+    UNSPECIFIED, // Equivalent to theme being omitted
+    STANDARD,
+    CHRISTMAS,
+}
+
 export interface Cashlink {
     address: string; // Userfriendly address
     message: string;
+    value: number;
     status: CashlinkState;
-}
-
-export interface CashlinkRequest extends BasicRequest {
-    senderAddress?: string;
-    senderBalance?: number;
-    cashlinkAddress?: string;
+    theme: CashlinkTheme;
+    cashlink?: string;
```

Comment:
I'd call this `url` or `link`, but not `cashlink`, because the container is already called `Cashlink`. This now means that users of this type have to write `cashlink.cashlink` in their code.

Reply:
I'll change it to `link` and rename `returnCashlink` to `returnLink` in the request accordingly.